test/distributed/org/apache/cassandra/distributed/impl/AbstractCluster.java

```diff
@@ -332,29 +333,12 @@ private IInvokableInstance newInstance()
             ++generation;
             IClassTransformer transformer = classTransformer == null ? null : classTransformer.initialise();
             ClassLoader classLoader = new InstanceClassLoader(generation, config.num(), version.classpath, sharedClassLoader, sharedClassPredicate, transformer);
-            Consumer<Throwable> stabilityInspector;
```

Comment:
this is problematic as this logic is shared with UpgradeCluster, which is expected to support forward and previous versions.  This logic tried to bring `JVMStabilityInspector` into the jvm-dtest API space.  

`org.apache.cassandra.distributed.api.IInstance#uncaughtException` was added specifically to avoid this.